store/tikv/region_request.go

```diff
@@ -178,6 +192,27 @@ func (s *RegionRequestSender) sendReqToRegion(bo *Backoffer, ctx *RPCContext, re
 	return
 }
 
+func (s *RegionRequestSender) getStoreToken(storeID uint64) error {
+	s.storeLimit.Lock()
```

Comment:
I think the lock is too coarse. Maybe we can define a guard field in the store struct and use atomic to reduce contention.

Reply:
Yes, I have tried to reconstruct the code to decrease granularity of the lock